### PR TITLE
fix(profiles): normalize community catalog paths before dedupe

### LINKED ISSUE #107: Normalize community catalog paths before uniqueness checks
## Context
Gemini review on merged PR #95 flagged a path normalization gap in catalog deduplication.

## Problem
Path uniqueness currently relies on string conversion, which can miss equivalent paths containing redundant components like `./`.

## Scope
`crates/clawcrate-profiles/src/lib.rs`

## Acceptance Criteria
- Normalize relative path components (at least `.`) before uniqueness insertion.
- Compare using normalized `PathBuf` keys rather than lossy string representation.
- Keep current restrictions for absolute and parent-dir escaping paths.
- Add tests for duplicate variants (`foo.yaml` vs `./foo.yaml`).

## References
- PR #95
- https://github.com/manuelpenazuniga/ClawCrate/pull/95#discussion_r3105831097
- Related epic: #7

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary
- normalize relative catalog paths before uniqueness checks to collapse redundant `./` components
- dedupe paths using normalized `PathBuf` keys instead of lossy string conversions
- preserve existing validation for absolute paths and parent-directory escape attempts
- add regression coverage for duplicate variants (`one.yaml` vs `./one.yaml`)

## Testing
- cargo fmt --check
- cargo test -p clawcrate-profiles

Closes #107

### CODE REVIEW

File: crates/clawcrate-profiles/src/lib.rs [hunk 1]
```diff
@@ -214,15 +214,15 @@ impl ProfileResolver {
...
+            let normalized_path = normalize_catalog_relative_path(&entry.path);
+            if !seen_paths.insert(normalized_path.clone()) {
                 return Err(ProfileError::InvalidCatalog {
                     path: catalog_path.to_path_buf(),
                     reason: format!("duplicate profile path '{}'", entry.path.display()),
                 });
             }
 
-            let profile_path = catalog_dir.join(&entry.path);
+            let profile_path = catalog_dir.join(&normalized_path);
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-profiles/src/lib.rs:225]: ok

File: crates/clawcrate-profiles/src/lib.rs [hunk 2]
```diff
@@ -543,6 +543,17 @@ fn normalize_string(input: &str) -> String {
...
+fn normalize_catalog_relative_path(path: &Path) -> PathBuf {
+    let mut normalized = PathBuf::new();
+    for component in path.components() {
+        if matches!(component, Component::CurDir) {
+            continue;
+        }
+        normalized.push(component.as_os_str());
+    }
+    normalized
+}
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-profiles/src/lib.rs:555]: okok
